apps/web/src/screens/HomePage/sections/Membership/index.tsx

```diff
@@ -0,0 +1,148 @@
+import { Element } from "react-scroll";
+import styled, { AnyStyledComponent } from "styled-components";
+
+import Icon from "react-eva-icons";
+
+import { CheckoutForm } from "../../../../components/CheckoutForm";
+import { PageConstraint } from "../../../../components/PageConstraint";
+import {
+  MeExpirationQueryHookResult,
+  MembershipTypes,
+  useMeExpirationQuery
+} from "../../../../generated/graphql";
+import { ConfirmationContainer } from "./ConfirmationContainer";
+import { Header, TierContainer } from "./TierContainer";
+
+export const ME_EXPIRATION_QUERY: any = gql`
+  query MeExpiration {
+    me {
+      membershipExpiration
+    }
+  }
+`;
+
+const MembershipWrapper: AnyStyledComponent = styled.div`
+  margin: auto;
+  margin-bottom: 100px 
+  width: 100%;
+  max-width: 1200px
+  padding: 0 5%;
+  font-family: "Nunito Sans"
+`;
+
+const Heading: AnyStyledComponent = styled.h1`
+  font-family: "Roboto", sans-serif;
+  text-transform: uppercase;
+  font-size: 23px;
+  & i {
+    margin-right: 10px;
+  }
+`;
+
+const Description: AnyStyledComponent = styled.p`
+  margin-bottom: 20px;
+  font-size: 19px;
+`;
+
+const Line: AnyStyledComponent = styled.hr`
+  border: 0;
+  height: 0;
+  border-top: 1px solid rgba(0, 0, 0, 0.1);
+  border-bottom: 1px solid rgba(255, 255, 255, 0.3);
+  margin: 100px 0;
+`;
+
+const TierList: AnyStyledComponent = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+
+  @media all and (min-width: 790px) {
+    flex-direction: row;
+  }
+`;
+
+type voidFunction = () => void;
+const Membership: React.FC = (): JSX.Element => {
+  const [tag, setTag]: [
+    MembershipTypes | undefined,
+    React.Dispatch<React.SetStateAction<MembershipTypes | undefined>>
+  ] = useState<MembershipTypes | undefined>(undefined);
+  const modalStyle: any = {
+    content: {
+      padding: "0",
+      minWidth: "350px",
+      width: "380px",
+      height: "380px",
+      top: "50%",
+      left: "50%",
+      right: "auto",
+      bottom: "auto",
+      marginRight: "-50%",
+      transform: "translate(-50%, -50%)"
+    }
+  };
+
+  const result: MeExpirationQueryHookResult = useMeExpirationQuery();
+  let expirationDate: string | undefined;
+
+  if (result.data && result.data.me) {
+    expirationDate = result.data.me.membershipExpiration;
+    result.data.me.membershipExpiration;
+  }
+
+  const removeTag: () => void = (): void => {
+    setTag(undefined);
+  };
+
+  const setYearly: voidFunction = (): void => setTag(MembershipTypes.Yearly);
+  const setSemesterly: voidFunction = (): void =>
+    setTag(MembershipTypes.Semesterly);
+
+  return (
+    <Element name="membership">
+      <MembershipWrapper>
+        <Line />
+        <Modal
+          isOpen={tag !== undefined}
+          onRequestClose={removeTag}
+          contentLabel="Example Modal"
+          style={modalStyle}
+        >
+          <Header>{tag}</Header>
+          <CheckoutForm tag={tag!} />
```

Comment:
The exclamation here doesn't stop there element from becoming undefined, just removes the type, seems like a hack since it can clearly be undefined, just not while the modal is open

Reply:
Thats the point. It doesn't matter whats show in the if modal is closed, but when it is open we know that tag must not be undefined. I can add the check if we want the extra safety